app/src/main/java/net/bible/service/db/readingplan/ReadingPlanDBAdapter.kt

```diff
@@ -0,0 +1,770 @@
+ *
+ */
+
+package net.bible.service.db.readingplan
+
+import android.content.ContentValues
+import android.provider.BaseColumns
+import android.util.Log
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.SerializationException
+import kotlinx.serialization.json.JSON
+import net.bible.android.BibleApplication
+import net.bible.android.activity.R
+import net.bible.android.view.activity.readingplan.DailyReading
+import net.bible.service.common.CommonUtils
+import net.bible.service.db.CommonDatabaseHelper
+import net.bible.service.db.readingplan.ReadingPlanDatabaseDefinition.DB_FALSE_VALUE
+import net.bible.service.db.readingplan.ReadingPlanDatabaseDefinition.DB_TRUE_VALUE
+import net.bible.service.db.readingplan.ReadingPlanDatabaseDefinition.ReadingPlanDays
+import net.bible.service.db.readingplan.ReadingPlanDatabaseDefinition.ReadingPlanMeta
+import net.bible.service.readingplan.PassageReader
+import org.apache.commons.lang3.StringUtils
+import org.crosswire.jsword.passage.Key
+import org.crosswire.jsword.versification.Versification
+import org.crosswire.jsword.versification.system.SystemNRSVA
+import org.crosswire.jsword.versification.system.Versifications
+import java.text.SimpleDateFormat
+import java.util.Calendar
+import java.util.Date
+import kotlin.collections.ArrayList
+
+/**
+ * @author Timmy Braun [tim.bze at gmail dot com] (2/14/2019)
+ */
+class ReadingPlanDBAdapter {
+    companion object {
+        const val TAG = "ReadingPlanDBAdapter"
+        const val INCLUSIVE_VERSIFICATION = SystemNRSVA.V11N_NAME
+        val app = BibleApplication.application
+        val dbHelper: CommonDatabaseHelper = CommonDatabaseHelper.getInstance()
+
+        private val dbWritable = dbHelper.writableDatabase
+        private val dbReadable = dbHelper.readableDatabase
+
+        private val dateBasedFormatMonthDay = SimpleDateFormat("MMM-d")
+        val dateBasedFormatWithYear = SimpleDateFormat("MMM-d/yyyy")
+
+    }
+
+    var metaCurrentActiveReadingPlanID: Int?
+        get() {
+            val projection = arrayOf(BaseColumns._ID)
+            val selection = "${ReadingPlanMeta.COLUMN_LAST_USED_PLAN}=?"
+            val selectionArgs = arrayOf(DB_TRUE_VALUE)
+            val cursor = dbReadable.query(
+                ReadingPlanMeta.TABLE_NAME,
+                projection,
+                selection,
+                selectionArgs,
+                null,
+                null,
+                null)
+            val result =
+                if (cursor.moveToFirst())
+                    cursor.getInt(0)
+                else null
+
+            cursor.close()
+            return result
+        }
+        set(newValue) {
+            if (newValue != null && newValue != 0) {
+                val setFalseValues = ContentValues().apply {
+                    put(ReadingPlanMeta.COLUMN_LAST_USED_PLAN, DB_FALSE_VALUE)
+                }
+                val setTheTrueValue = ContentValues().apply {
```

Comment:
inconsistency in variable naming. Maybe

```suggestion
                val setTrueValues = ContentValues().apply {
```

Reply:
Well my reason for the name is: setFalseValues (first mark every row as false). setTheTrueValue (Set the ONE reading plan as true which was last used)

Reply:
Okay, I see...

Reply:
Essentially you are wanting to store which reading plan is active, right? Then I think you should write somewhere just the index of that entry, instead of boolean value on each column.

Reply:
If you change it like that, this property will be a lot simpler and cleaner code. I'm thinking that most likely you could also make this non-nullable (with a sane default value, like 0).

Reply:
I think that's a good idea. Fixed.